src/optimizer/remove_unused_columns.cpp

```diff
@@ -56,9 +56,15 @@ void RemoveUnusedColumns::VisitOperator(LogicalOperator &op) {
 	switch (op.type) {
 	case LogicalOperatorType::LOGICAL_AGGREGATE_AND_GROUP_BY: {
 		// aggregate
+		auto &aggr = op.Cast<LogicalAggregate>();
+		// if there is more than one grouping set, the group by most likely has a rollup or cube
+		// If there is an equality join underneath the aggregate, this can change the groups to avoid unused columns
+		// This causes the duplicate eliminator to ignore functionality provided by grouping sets
+		if (aggr.grouping_sets.size() > 1) {
+			return;
```

Comment:
Is immediately returning not a bit too strict here? I think we could still recurse into children with `is_root` set to `true`.